simulator/src/main/java/peersim/kademlia/KademliaProtocol.java

```diff
@@ -182,18 +181,23 @@ private void handleResponse(Message m, int myPid) {
       // Update the find operation record with the closest set of neighbors received
       fop.elaborateResponse((BigInteger[]) m.body);
       fop.addMessage(m.id);
-      logger.info("Handleresponse FindOperation " + fop.getId() + " " + fop.getAvailableRequests());
+
       // Save received neighbour in the closest Set of fin operation
       BigInteger[] neighbours = (BigInteger[]) m.body;
       if (callback != null) callback.nodesFound(fop, neighbours);
-      for (BigInteger neighbour : neighbours) routingTable.addNeighbour(neighbour);
+      for (BigInteger neighbour : neighbours)
+        if (Util.nodeIdtoNode(neighbour, myPid).getKademliaProtocol().getKademliaNode().isServer())
+          routingTable.addNeighbour(neighbour);
 
       if (!fop.isFinished()
           && Arrays.asList(neighbours).contains(fop.getDestNode())
           && !(fop instanceof RegionBasedFindOperation)) {
 
         logger.warning("Found node " + fop.getDestNode());
-
+        if (fop instanceof PutOperation) {

```

Comment:
fop might suggest it's a find operation

Reply:
well, technically it is, since get and put operation extends from find operation.